addons/account_asset/account_asset.py

```diff
@@ -1,351 +1,325 @@
+    account_asset_id = fields.Many2one('account.account', string='Asset Account', required=True, domain=[('type', '=', 'other')])
+    account_income_recognition_id = fields.Many2one('account.account', string='Recognition Income Account', domain=[('type', '=', 'other')])
+    account_depreciation_id = fields.Many2one('account.account', string='Depreciation Account', required=True, domain=[('type', '=', 'other')])
+    journal_id = fields.Many2one('account.journal', string='Journal', required=True)
+    company_id = fields.Many2one('res.company', string='Company', required=True, default=lambda self: self.env['res.company']._company_default_get('account.asset.category'))
+    method = fields.Selection([('linear', 'Linear'), ('degressive', 'Degressive')], string='Computation Method', required=True, default='linear', help="Choose the method to use to compute the amount of depreciation lines.\n"
+        "  * Linear: Calculated on basis of: Gross Value / Number of Depreciations\n"
+        "  * Degressive: Calculated on basis of: Residual Value * Degressive Factor")
+    method_number = fields.Integer(string='Number of Depreciations', default=5, help="The number of depreciations needed to depreciate your asset")
+    method_period = fields.Integer(string='Period Length', default=1, help="State here the time between 2 depreciations, in months", required=True)
+    method_progress_factor = fields.Float('Degressive Factor', default=0.3)
+    method_time = fields.Selection([('number', 'Number of Depreciations'), ('end', 'Ending Date')], string='Time Method', required=True, default='number',
+        help="Choose the method to use to compute the dates and number of depreciation lines.\n"
+           "  * Number of Depreciations: Fix the number of depreciation lines and the time between 2 depreciations.\n" \
+           "  * Ending Date: Choose the time between 2 depreciations and the date the depreciations won't go beyond.")
+    method_end = fields.Date('Ending date')
+    prorata = fields.Boolean(string='Prorata Temporis', help='Indicates that the first depreciation entry for this asset have to be done from the purchase date instead of the first January')
+    open_asset = fields.Boolean(string='Skip Draft State', help="Check this if you want to automatically confirm the assets of this category when created by invoices.")
+    type = fields.Selection([('sale', 'Sale: Revenue Recognition'), ('purchase', 'Purchase: Asset')], required=True, index=True, default='purchase')
+
+    @api.onchange('account_asset_id')
+    def onchange_account_asset(self):
+        self.account_depreciation_id = self.account_asset_id
+
+    @api.onchange('type')
+    def onchange_type(self):
+        if self.type == 'sale':
+            self.prorata = True
+            self.method_period = 1
+        else:
+            self.method_period = 12
 
 
-class account_asset_asset(osv.osv):
+class account_asset_asset(models.Model):
```

Comment:
@mdi-odoo FYI: repeated exclamations marks are considered as aggressive in an online discussion...